subprojects/tooling-api/src/main/java/org/gradle/tooling/ProjectConnection.java

```diff
@@ -155,6 +158,17 @@
      */
     BuildActionExecuter.Builder action();
 
+    /**
+     * Notifies all daemons about file changes made by an external process, like an IDE.
+     *
+     * The daemons will use this information to update the retained file system state.
+     *
+     * @param locations Absolute paths which have been changed by the external process.
+     * @since 6.1
+     */
+    @Incubating
+    void notifyDaemonsAboutChangedPaths(List<String> locations);
```

Comment:
I think we should clarify a few things about the API, in the javadoc or perhaps by tweaking the signature:

- What kinds of exceptions are thrown?
- Is the operation synchronous or asynchronous? If asynchronous, how do I find out what happened?
- What do the paths mean? Are they absolute? relative? if so, to what?
- What should I pass when I rename something? delete something? create something?
- Are there any delivery guarantees if I start a build or some other operation after calling this method? that is, if I call this method and then immediately start a build, what happens?
- What happens if the version of Gradle I'm using does not support this feature?

Reply:
I clarified the Javadoc.